src/cargo/ops/cargo_rustc/mod.rs

```diff
@@ -599,8 +599,10 @@ fn build_base_args(cx: &Context,
         cmd.arg("-g");
     }
 
-    if ndebug {
-        cmd.args(&["--cfg", "ndebug"]);
+    if debug_assertions && opt_level > 0 {
```

Comment:
What's the reasoning for this rather than, say, `format!("debug-assertions={}", if debug_assertions { "on" } else { "off" })`?